activerecord/test/cases/adapter_test.rb

```diff
@@ -485,11 +485,17 @@ def setup
     end
 
     def test_truncate
+      posts_attributes = Post.all.map(&:attributes)
+
       assert_operator Post.count, :>, 0
 
       @connection.truncate("posts")
 
       assert_equal 0, Post.count
+    ensure
```

Comment:
Would it make more sense to do this instead?

```
ActiveRecord::FixtureSet.reset_cache
ActiveRecord::FixtureSet.create_fixtures(FIXTURES_ROOT, "posts")
```

Reply:
seems like a good idea, and the way to extend for the other tests that truncate tables.

Reply:
I've gone ahead and implemented the suggested approach on other tests as well, thanks!